entoas/extension.go

```diff
@@ -0,0 +1,196 @@
+	Extension struct {
+		entc.DefaultExtension
+		config    *Config
+		mutations []MutateFunc
+		out       io.Writer
+	}
+	// ExtensionOption allows managing Extension configuration using functional arguments.
+	ExtensionOption func(*Extension) error
+	// MutateFunc defines a closure to be called on a generated spec.
+	MutateFunc func(*gen.Graph, *spec.Spec) error
+)
+
+// NewExtension returns a new entoas extension with default values.
+func NewExtension(opts ...ExtensionOption) (*Extension, error) {
+	ex := &Extension{config: &Config{DefaultPolicy: policy.Expose}}
+	for _, opt := range opts {
+		if err := opt(ex); err != nil {
+			return nil, err
+		}
+	}
+	return ex, nil
+}
+
+// Hooks of the Extension.
+func (ex *Extension) Hooks() []gen.Hook {
+	return []gen.Hook{ex.generate}
+}
+
+// Annotations of the extensions.
+func (ex *Extension) Annotations() []entc.Annotation {
+	return []entc.Annotation{ex.config}
+}
+
+// DefaultPolicy sets the default ExclusionPolicy to use of none is given on a (sub-)schema.
+func DefaultPolicy(p policy.Policy) ExtensionOption {
+	return func(ex *Extension) error {
+		ex.config.DefaultPolicy = p
+		return nil
+	}
+}
+
+// Mutations adds the given mutations to the spec generator.
+//
+// A MutateFunc is a simple closure that can be used to edit the generated spec. It can be used to add custom endpoints
+// or alter the spec in any other way.
+func Mutations(ms ...MutateFunc) ExtensionOption {
+	return func(ex *Extension) error {
+		ex.mutations = append(ex.mutations, ms...)
+		return nil
+	}
+}
+
+// SimpleModels enables the simple model generation feature.
+//
+// The OAS generator by default creates one view per endpoint. The naming strategy is the following:
+// - <S><Op> for 1st level operation Op on schema S
+// - <S><Op>_<E> for an eager loaded edge E on 1st level operation Op on schema S
+// - <S>_<E><Op> for a 2nd level operation Op on edge E on schema S
+//
+// By enabling she SimpleModels configuration the generator simply adds the defined schemas with all fields and edges.
+// Serialization groups have no effects in this mode.
+func SimpleModels() ExtensionOption {
+	return func(ex *Extension) error {
+		ex.config.SimpleModels = true
+		return nil
+	}
+}
+
+// SpecTitle sets the title of the Info block.
+func SpecTitle(v string) ExtensionOption {
+	return Mutations(func(_ *gen.Graph, spec *spec.Spec) error {
+		spec.Info.Title = v
+		return nil
+	})
+}
+
+// SpecDescription sets the title of the Info block.
+func SpecDescription(v string) ExtensionOption {
+	return Mutations(func(_ *gen.Graph, spec *spec.Spec) error {
+		spec.Info.Description = v
+		return nil
+	})
+}
+
+// SpecVersion sets the version of the Info block.
+func SpecVersion(v string) ExtensionOption {
+	return Mutations(func(_ *gen.Graph, spec *spec.Spec) error {
+		spec.Info.Version = v
+		return nil
+	})
+}
+
+// DumpTo dumps the current specs content to the given io.Writer.
+func DumpTo(out io.Writer) ExtensionOption {
```

Comment:
Why not `WriteTo`?

Reply:
`Write` to me is when you have bytes written unchanged. `Dump` if there is some kind of mutation or conversion of the data. In this case `Spec` to  `[]byte`. I can change it though.